pagespeed/kernel/http/http_names.cc

```diff
@@ -189,43 +193,138 @@ const char* HttpStatus::GetReasonPhrase(HttpStatus::Code rc) {
   return "";
 }
 
-StringPieceVector HttpAttributes::SortedHopByHopHeaders() {
-  // http://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html
-  const int kReserveSize = 10;
-  int index = 0;
-  StringPieceVector names(kReserveSize);
-
-  // This exact mechanism of initializing the names is used because it allows
-  // populating a StringPieceVector from locally defined string constants
-  // without having runtime calls to strlen to find the length.
-  names[index++] = StringPiece(HttpAttributes::kConnection);
-  names[index++] = StringPiece(HttpAttributes::kKeepAlive);
-  names[index++] = StringPiece(HttpAttributes::kProxyAuthenticate);
-  names[index++] = StringPiece(HttpAttributes::kProxyAuthorization);
-  names[index++] = StringPiece(HttpAttributes::kSetCookie);
-  names[index++] = StringPiece(HttpAttributes::kSetCookie2);
-  names[index++] = StringPiece(HttpAttributes::kTE);
-  names[index++] = StringPiece(HttpAttributes::kTrailers);
-  names[index++] = StringPiece(HttpAttributes::kTransferEncoding);
-  names[index++] = StringPiece(HttpAttributes::kUpgrade);
-  DCHECK_EQ(kReserveSize, index);
-  return names;
+namespace {
+
+class EndToEndHeadersContainer {
+ public:
+  EndToEndHeadersContainer() {
+    BuildEndToEnd();
+    BuildHopByHop();
+    BuildCachingHeadersToBeRemoved();
+  }
+
+  void BuildEndToEnd() {
+    // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html#sec13.5.1
```

Comment:
Not worth changing, but: you may want to start using the new HTTP1.1 RFCs (RFC7230-RFC7235), not the obsolete RFC2616, as they should be more precise.